### PR TITLE
[13.0][ADD] sale_exception_financial_risk: Glue module between sale_exception and sale_financial_risk

### PR DESCRIPTION
**Pull request by victoralmau**
Glue module between `sale_exception` and `sale_financial_risk`.

Please @pedrobaeza and @joao-p-marques can you review it?

@Tecnativa TT30755

### CODE REVIEW

File: sale_exception_financial_risk/models/sale_order.py [hunk 1]
```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Tecnativa - Víctor Martínez
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+from odoo import api, fields, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    risk_amount = fields.Monetary(
```
Comment (pedrobaeza) [inline sale_exception_financial_risk/models/sale_order.py]: Why do you need this field and aren't using `sale_financial_risk` stuff?

File: sale_exception_financial_risk/models/sale_order.py [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+                item.company_id.currency_id,
+                item.company_id,
+                item.date_order
+                and item.date_order.date()
+                or fields.Date.context_today(item),
+                round=False,
+            )
+
+    def action_confirm(self):
+        if not self.env.context.get("bypass_risk", False):
```
Comment (pedrobaeza) [inline sale_exception_financial_risk/models/sale_order.py]: This shouldn't be needed, as sale_exception rules should be checked automatically.

File: sale_exception_financial_risk/data/sale_exception_data.xml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <record model="exception.rule" id="sale_financial_risk_exceeds_sale_order_risk">
+            <field name="name">This sale order exceeds the sales orders risk</field>
+            <field
+                name="description"
+            >This sale order exceeds the sales orders risk</field>
+            <field name="active" eval="True" />
+            <field name="sequence" eval="40" />
+            <field name="model">sale.order</field>
+            <field name="code">
+risk_amount = obj.currency_id._convert(
```
Comment (pedrobaeza) [inline sale_exception_financial_risk/data/sale_exception_data.xml]: It should be only one exception, and use `evaluate_risk_message` method

File: sale_exception_financial_risk/data/sale_exception_data.xml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    obj.company_id.currency_id,
+    obj.company_id,
+    obj.date_order
+    and obj.date_order.date()
+    or context_today(),
+    round=False,
+)
+if obj.partner_id.risk_sale_order_limit and (
+    (obj.partner_id.risk_sale_order + risk_amount) > obj.partner_id.risk_sale_order_limit
+): failed=True</field>
```
Comment (pedrobaeza) [inline sale_exception_financial_risk/data/sale_exception_data.xml]: Remember to not use this way, but direct expression assignation with `bool` if needed.

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Why 2 exceptions? It should be the same IMO

**Comment by victoralmau** [comment]
Thanks @pedrobaeza, changes done.

**Review by pedrobaeza — APPROVED** [review]

**Review by ernestotejeda — APPROVED** [review]

File: sale_exception_financial_risk/README.rst [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+.. |badge4| image:: https://img.shields.io/badge/weblate-Translate%20me-F47D42.png
+    :target: https://translation.odoo-community.org/projects/credit-control-13-0/credit-control-13-0-sale_exception_financial_risk
+    :alt: Translate me on Weblate
+.. |badge5| image:: https://img.shields.io/badge/runbot-Try%20me-875A7B.png
+    :target: https://runbot.odoo-community.org/runbot/262/13.0
+    :alt: Try me on Runbot
+
+|badge1| |badge2| |badge3| |badge4| |badge5| 
+
+Glue module between sale_exception and sale_financial_risk (in OCA/credit-control)
```
Comment (joao-p-marques) [inline sale_exception_financial_risk/README.rst:28]: Can you provide a bit more information in this description?

File: sale_exception_financial_risk/data/sale_exception_data.xml [hunk 6]
```diff
@@ -0,0 +1,15 @@
+<?xml version="1.0" encoding="utf-8" ?>
+<odoo>
+    <data noupdate="1">
+        <record model="exception.rule" id="sale_financial_risk_have_risk_message">
+            <field name="name">This sale order have risk message</field>
```
Comment (joao-p-marques) [inline sale_exception_financial_risk/data/sale_exception_data.xml]:
```suggestion
            <field name="name">This sale order has risk messages</field>
```

File: sale_exception_financial_risk/data/sale_exception_data.xml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+            <field name="description">This sale order have risk message</field>
```
Comment (joao-p-marques) [inline sale_exception_financial_risk/data/sale_exception_data.xml]: Same has above

**Review by joao-p-marques — COMMENTED** [review]
Code review :+1: 

Just some small comments, otherwise looks good

**Review by joao-p-marques — APPROVED** [review]

**Comment by joao-p-marques** [comment]
Well, the logs problem again... Let's wait to see if https://github.com/odoo/odoo/pull/75387 gets merged so we can unblock the CIs here...

**Pull request merged**